### PR TITLE
Reduced cognitive complexity

### PR DESCRIPTION
**Pull request by mrickard**
## Proposed Release Notes
Reduced cognitive complexity in lib/metrics/normalizer.js

## Links
https://issues.newrelic.com/browse/NEWRELIC-5292

## Details

Closes NEWRELIC-5292

### CODE REVIEW

**Comment by mrickard** [comment]
Draft PR because it makes sense to roll up several of these in one PR.

File: lib/metrics/normalizer.js [hunk 1]
```diff
@@ -112,45 +110,53 @@ MetricNormalizer.prototype.load = function load(json) {
   }
 }
 
+/**
+ *  Helper for loadFromConfig
+ *
+ * @param rule {object} from this.config.rules
+ * @param ctx {object} context from the calling function
+ */
+const processNameRule = function processName(rule, ctx) {
```
Comment (jordigh) [inline lib/metrics/normalizer.js:119]: Is using an assignment to rename a function something that is less cognitively complex?

File: lib/metrics/normalizer.js [hunk 2]
```diff
@@ -112,45 +110,53 @@ MetricNormalizer.prototype.load = function load(json) {
...
+    ctx.rules.splice(insert, 0, new NormalizerRule(json))
+  }
+}
 /**
  * Load any rules found in the configuration into a metric normalizer.
  *
  * Operates via side effects.
  */
 MetricNormalizer.prototype.loadFromConfig = function loadFromConfig() {
+  const ctx = this
```
Comment (jordigh) [inline lib/metrics/normalizer.js:155]: I'd just use `this` and invoke the function with `apply` in order to save ourselves one more variable to think about, but that's just me.

File: lib/metrics/normalizer.js (hunk 1 above)
Comment (mrickard) [reply lib/metrics/normalizer.js:119]: Apparently so, oddly enough.

File: lib/metrics/normalizer.js (hunk 2 above)
Comment (mrickard) [reply lib/metrics/normalizer.js:155]: Ordinarily, I'd reach for `.bind(this)`, and `.apply` is similar, but context is still getting lost in the `forEach` with those. I'm not super enthusiastic about passing it in as an argument, but it's proving to be more reliable than `bind` or `apply` here.

File: lib/metrics/normalizer.js (hunk 1 above)
Comment (mrickard) [reply lib/metrics/normalizer.js:119]: Ah, wait--I see what you mean. I'd thought you were referring to pulling the processing block into a function. The cognitive complexity total is mostly affected by breaks in flow and nesting. Assignment doesn't seem to move the needle, but I can take it out if it's jarring.

Comment (jordigh) [reply lib/metrics/normalizer.js]: Thank you, yes, that's better!

**Review by jordigh — APPROVED** [review]

**Pull request merged**